packages/sdk/contractkit/src/wrappers/FederatedAttestations.ts

```diff
@@ -1,8 +1,170 @@
+   * @return countsPerIssuer Array of number of identifiers returned per issuer
+   * @return identifiers Array (length == sum([0])) of identifiers
+   * @dev Adds identifier info to the arrays in order of provided trustedIssuers
+   * @dev Expectation that only one attestation exists per (identifier, issuer, account)
+   */
+  lookupIdentifiers: (
+    account: Address,
+    trustedIssuers: Address[]
+  ) => Promise<{
+    countsPerIssuer: string[]
+    identifiers: string[]
+    '0': string[]
+    '1': string[]
+  }> = proxyCall(this.contract.methods.lookupIdentifiers)
+
+  /**
+   * @notice Helper function for lookupAttestations to calculate the total number of
+   * attestations completed for an identifier by each trusted issuer
+   * @param identifier Hash of the identifier
+   * @param trustedIssuers Array of n issuers whose attestations will be included
+   * @return totalAttestations Sum total of attestations found
+   * @return countsPerIssuer Array of number of attestations found per issuer
+   */
+  lookupAttestations: (
+    identifier: Identifier,
+    trustedIssuers: Address[]
+  ) => Promise<{
+    countsPerIssuer: string[]
+    accounts: string[]
+    signers: string[]
+    issuedOns: string[]
+    publishedOns: string[]
+    '0': string[]
+    '1': string[]
+    '2': string[]
+    '3': string[]
+    '4': string[]
+  }> = proxyCall(this.contract.methods.lookupAttestations)
+
+  /**
+   * @notice Registers an attestation directly from the issuer
+   * @param identifier Hash of the identifier to be attested
+   * @param account Address of the account being mapped to the identifier
+   * @param issuedOn Time at which the issuer issued the attestation in Unix time
+   * @dev Attestation signer and issuer in storage is set to msg.sender
+   * @dev Throws if an attestation with the same (identifier, issuer, account) already exists
+   */
+  registerAttestationAsIssuer: (
+    identifier: Identifier,
+    account: Address,
+    issuedOn: number | string
+  ) => CeloTransactionObject<void> = proxySend(
+    this.connection,
+    this.contract.methods.registerAttestationAsIssuer
+  )
+
+  /**
+   * @notice Registers an attestation with a valid signature
+   * @param identifier Hash of the identifier to be attested
+   * @param issuer Address of the attestation issuer
+   * @param account Address of the account being mapped to the identifier
+   * @param issuedOn Time at which the issuer issued the attestation in Unix time
+   * @param signer Address of the signer of the attestation
+   * @param v The recovery id of the incoming ECDSA signature
+   * @param r Output value r of the ECDSA signature
+   * @param s Output value s of the ECDSA signature
+   * @dev Throws if an attestation with the same (identifier, issuer, account) already exists
+   */
+  registerAttestation: (
+    identifier: Identifier,
+    issuer: Address,
+    account: Address,
+    signer: Address,
+    issuedOn: number | string,
+    v: number | string,
```

Comment:
it would be nice if we could pass in a `Signature` object here instead :) example: https://github.com/celo-org/celo-monorepo/blob/1acca063339adaac710341279d3ccfaa029e4b0a/packages/sdk/contractkit/src/wrappers/Accounts.ts#L162-L175

Reply:
Good idea!

Reply:
After implementing, I've noticed you can't call `.send()` directly but instead have to create the Object then submit the transaction.

Reply:
Good point Izzy. I think we should definitely strive to use the Signature type when we invoke this function, and when I first read your comment I agreed completely. I also see Jesse's point though on how keeping it this way allows us to pass the parameters through to proxySend() without any processing, which is a little cleaner IMO so I might also be in favor of keeping it this way and parsing v,r,s out one level up. Wdyt?

Reply:
sounds good! We can use the Signature type on the sdk side instead :)